desafio2.js

```diff
@@ -67,11 +70,41 @@ throw new Error(error)        }
         }catch(error){
             throw new Error
         }
-}}
+}
+    async putByID(id,obj){
+        try {
+            const data=await fs.promises.readFile(this.archivo,"utf-8");
+            const jsonData=JSON.parse(data)
+            if(id<=jsonData.length){
```

Comment:
se puede agregar una condicion de si existe el producto a modificar en el array, pero para fines practicos esta condicion no es necesario, en el try pondría directamente:
try {
            const data=await fs.promises.readFile(this.archivo,"utf-8");
            const jsonData=JSON.parse(data)
            const product = jsonData.find(i=>i.id==id)
            const prodModify = {...product, ...obj}
            const products = jsonData.filter(i=>i.id!=id)
            products.push(prodModify)
            await fs.promises.writeFile(this.archivo,JSON.stringify(products,null,2))
            return jsonData
        }